File: app/lib/model/category.dart
class Category {
  String id;
  String name;
  String img;

  Category({this.id, this.name, this.img});

  Category.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        img = json['img'];

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'img': img,
  };
}

File: be/internal/entity/category.go
package entity

import "github.com/google/uuid"

type Category struct {
	Base

	Name        string  `gorm:"type:varchar(128); not null" validate:"required,max=128" json:"name"`
	Img 		string `gorm:"type:varchar(10485760); not null" json:"img"`
	UserId 		uuid.UUID  `gorm:"type:uuid; not null" json:"userId"`
}